airflow/models/taskinstance.py

```diff
@@ -473,7 +473,7 @@ def clear_task_instances(
                 ti.refresh_from_task(task)
                 if TYPE_CHECKING:
                     assert ti.task
-                ti.max_tries = ti.try_number + task.retries
+                ti.max_tries = ti.try_number + (task.retries if task.retries is not None else 0)
```

Comment:
Oh, if you make it like this, it can be even simpler in Python:
```suggestion
                ti.max_tries = ti.try_number + task.retries or 0
```

Reply:
@jscheffl Thank you Jens for adding the suggestion, I have updated the code to use ti.max_tries = ti.try_number + (task.retries or 0)